### PR TITLE
[Ping] - documentation and code improvements for SaaS environments 

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-36504)

## Description
Return a custom error message when Ping isn't executed from a custom engine.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/ReleaseNotes/1_14_42.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+#### Scripts
+
+##### Ping
+
+- Added a custom error message indicating that the script can only run on custom engines in xsoar-saas and xsiam.
```
Comment (dantavori) [inline Packs/CommonScripts/ReleaseNotes/1_14_42.md:6]:
```suggestion
- Improved the error message to clarify that the script can only run on a custom engine on Cortex XSOAR 8 and Cortex XSIAM.
```

File: Packs/CommonScripts/Scripts/Ping/Ping.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ args:
   name: address
   predefined: []
   required: true
-comment: Pings an IP or url address, to verify it's up
+comment: Pings an IP or url address, to verify it's up, on xsoar8 and xsiam it can be used only on custom engines.
```
Comment (dantavori) [inline Packs/CommonScripts/Scripts/Ping/Ping.yml:8]:
```suggestion
comment: Pings an IP or url address, to verify it's up. Note: On Cortex XSOAR 8 and Cortex XSIAM, the script can run only on a custom engine.
```
please update the readme as well

File: Packs/CommonScripts/Scripts/Ping/Ping_test.py [hunk 3]
```diff
@@ -34,3 +34,21 @@ def test_fail_ping(mocker):
     # call_args last call with a tuple of args list and kwargs
     err_msg = return_error_mock.call_args[0][0]
     assert 'Name does not resolve' in err_msg or 'Name or service not known' in err_msg
+
+
+def test_fail_ping_permission_error_xsoar8(mocker):
```
Comment (dantavori) [inline Packs/CommonScripts/Scripts/Ping/Ping_test.py:39]: luckily is_xsoar_on_prem() returns false :) but it's ok

**Review by dantavori — APPROVED** [review]

File: Packs/CommonScripts/Scripts/Ping/Ping_test.py (hunk 3 above)
Comment (GuyAfik) [reply Packs/CommonScripts/Scripts/Ping/Ping_test.py:39]: true, i will mock it anyway

File: Packs/CommonScripts/Scripts/Ping/Ping.yml (hunk 2 above)
Comment (GuyAfik) [reply Packs/CommonScripts/Scripts/Ping/Ping.yml:8]: the script doesn't have a read me :/

**Review by JudahSchwartz — APPROVED** [review]

**Pull request merged**